### PR TITLE
fix: keep native range on cross-block dragging 

### PR DESCRIPTION
**Pull request by fundon**
Related:

* #564 
* #783 

## Before

https://user-images.githubusercontent.com/27926/214598325-a3fdcac6-e2d4-4da8-92bd-3230afe2a21b.mp4

## After

https://user-images.githubusercontent.com/27926/214597521-4f17688d-f408-4a77-ae97-e30f50bb65a9.mp4

## Others

Test cases will be added later

### CODE REVIEW

**Review by doodlewind — APPROVED** [review]
LGTM! I tweaked a bit, changing the `isDownward` naming to `isBackward`, which matches the mental model in Slate.

File: tests/selection.spec.ts [hunk 1]
```diff
@@ -649,3 +650,78 @@ test('the cursor should move to closest editor block when clicking outside conta
...
+  await assertClipItems(page, 'text/plain', '45');
+  // blur
+  await page.mouse.click(0, 0);
+
+  await dragBetweenIndices(
+    page,
+    [1, 2],
+    [1, 0],
+    { x: 0, y: 0 },
+    { x: -50, y: 0 }
```
Comment (himself65) [inline tests/selection.spec.ts:673]: why 50???

Comment (himself65) [reply tests/selection.spec.ts:673]: image

the target position is on red dot

Comment (fundon) [reply tests/selection.spec.ts:673]: see https://github.com/toeverything/blocksuite/pull/949

**Pull request merged**